### PR TITLE
Assignment-1

### PR DESCRIPTION
**Pull request by meta-ishita-mathur**

### CODE REVIEW

File: Assignments/arrayImpl/ArrayImpl.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package arrayImpl;
+
+class Stack
+{
+	int arr[];
+	int size,index=0;
```
Comment (sandeep-sky) [inline Assignments/arrayImpl/ArrayImpl.java]: Use the required spaces. Above line should be `int size, index = 0;`

File: Assignments/arrayImpl/ArrayImpl.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	
+	public Stack(int size)
+	{
+		this.size=size;
```
Comment (sandeep-sky) [inline Assignments/arrayImpl/ArrayImpl.java]: Space required `this.size=size;` should be `this.size = size;`.

File: Assignments/arrayImpl/ArrayImpl.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+		arr = new int[size];
+	}
+	
+	public boolean isFull()
+	{
+		if(index==size)
```
Comment (sandeep-sky) [inline Assignments/arrayImpl/ArrayImpl.java]: `index` always starts with zero. How `size` of array can be equal to `index`?

File: Assignments/arrayImpl/ArrayImpl.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+			return true;
+		else
+			return false;
+	}
+	public void push(int element)
+	{
+		if(isFull())
+		{
```
Comment (sandeep-sky) [inline Assignments/arrayImpl/ArrayImpl.java]: Curly braces not required for single statement inside `if` block.

File: Assignments/arrayImpl/ArrayImpl.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+			System.out.println("Stack overflow");
+		}
+		arr[index]=element;
+		index++;
+	}
+	
+	public boolean isEmpty()
+	{
+		if (index==0)
+			return true;
```
Comment (sandeep-sky) [inline Assignments/arrayImpl/ArrayImpl.java]: Avoid multiple `return` statements in a method.

File: Assignments/arrayImpl/ArrayImpl.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	}
+	
+	
+	public int pop()
+	{
+		if (isEmpty())
+		{
+			System.out.println("Stack is empty");
+		}
+		return arr[--index];
```
Comment (sandeep-sky) [inline Assignments/arrayImpl/ArrayImpl.java]: It will throw an error if stack is empty.

File: Assignments/assignment2/StackImpl.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+	private Node top;
+	public Stack()
+	{
+		this.top=null;
+	}
+	public void push(int x)
+	{
+		Node node = new Node();
+		System.out.println("Inserting"+x);
+		node.data=x;
```
Comment (sandeep-sky) [inline Assignments/assignment2/StackImpl.java]: Required spaces are mandatory for all the files.

File: Assignments/palindrome/PalindromeTest.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+		Scanner in=new Scanner(System.in);
+		String inputString = in.nextLine();
+		Stack<Character> stack = new Stack<Character>();
+		
+		for(int i=0;i<inputString.length();i++)
+		{
+			stack.push(inputString.charAt(i));
+    	}
+	
+    	String reverseString = "";
```
Comment (sandeep-sky) [inline Assignments/palindrome/PalindromeTest.java]: Variable should be declared at top of the class.

File: Assignments/shoppingCart/Cart.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    
+    public void addProduct(int pid,String name,double price,int quantity)
+    {
+    	int update = 0;
+		for(Product p : cartItems)
+		{
+			if(p.getName().equals(name))
+			{
+				update = 1;
+				int q = p.getQuantity();
```
Comment (sandeep-sky) [inline Assignments/shoppingCart/Cart.java]: Naming convention. `p, q, pid, update, getPid`

File: Assignments/shoppingCart/Main.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+			System.out.println("2. Remove item from cart");
+			System.out.println("3. Generate bill");
+			System.out.println("4. Exit");
+			System.out.println("Enter your choice:");
+			choice = sc.nextInt();
+		
+			switch(choice)
+			{
+				case 1: System.out.println("Enter the pid, name, price, quantity of the item");
+						int pid = sc.nextInt();
```
Comment (sandeep-sky) [inline Assignments/shoppingCart/Main.java]: Code indentation.

File: Assignments/shoppingCart/Product.java [hunk 11]
```diff
@@ -0,0 +1,73 @@
+package shoppingCart;
+
+public class Product 
+{
+	private int pid;
+	private String name;
+	private double price;
+	private int quantity;
+	
+	public Product()
```
Comment (sandeep-sky) [inline Assignments/shoppingCart/Product.java]: Why this constructor?

File: Assignments/assignment2/StackImpl.java (hunk 7 above)
Comment (meta-ishita-mathur) [reply Assignments/assignment2/StackImpl.java]: done

File: Assignments/palindrome/PalindromeTest.java (hunk 8 above)
Comment (meta-ishita-mathur) [reply Assignments/palindrome/PalindromeTest.java]: done

File: Assignments/shoppingCart/Cart.java [hunk 12]
```diff
@@ -0,0 +1,59 @@
+package shoppingCart;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Cart 
+{
+    List<Product> cartItems = new ArrayList<Product>();
```
[the remark this answers is not in the document]
Comment (meta-ishita-mathur) [reply Assignments/shoppingCart/Cart.java]: done

File: Assignments/shoppingCart/Cart.java [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+    
+    public void addProduct(int pid,String name,double price,int quantity)
+    {
+    	int update = 0;
+		for(Product p : cartItems)
```
[the remark this answers is not in the document]
Comment (meta-ishita-mathur) [reply Assignments/shoppingCart/Cart.java]: done

File: Assignments/shoppingCart/Cart.java (hunk 9 above)
Comment (meta-ishita-mathur) [reply Assignments/shoppingCart/Cart.java]: done

File: Assignments/shoppingCart/Product.java (hunk 11 above)
Comment (meta-ishita-mathur) [reply Assignments/shoppingCart/Product.java]: done

File: Assignments/shoppingCart/Main.java (hunk 10 above)
Comment (meta-ishita-mathur) [reply Assignments/shoppingCart/Main.java]: ddone

**Pull request merged**